examples/pokemon-service-common/src/lib.rs

```diff
@@ -242,59 +242,63 @@ pub async fn capture_pokemon(
             },
         ));
     }
-    let output_stream = stream! {
-        loop {
-            use std::time::Duration;
-            match input.events.recv().await {
-                Ok(maybe_event) => match maybe_event {
-                    Some(event) => {
-                        let capturing_event = event.as_event();
-                        if let Ok(attempt) = capturing_event {
-                            let payload = attempt.payload.clone().unwrap_or_else(|| CapturingPayload::builder().build());
-                            let pokeball = payload.pokeball().unwrap_or("");
-                            if ! matches!(pokeball, "Master Ball" | "Great Ball" | "Fast Ball") {
-                                yield Err(
+    let output_stream = FnStream::new(|tx| {
+        Box::pin(async move {
+            loop {
+                use std::time::Duration;
+                match input.events.recv().await {
+                    Ok(maybe_event) => match maybe_event {
+                        Some(event) => {
+                            let capturing_event = event.as_event();
+                            if let Ok(attempt) = capturing_event {
+                                let payload = attempt
+                                    .payload
+                                    .clone()
+                                    .unwrap_or_else(|| CapturingPayload::builder().build());
+                                let pokeball = payload.pokeball().unwrap_or("");
+                                if !matches!(pokeball, "Master Ball" | "Great Ball" | "Fast Ball") {
+                                    tx.send(Err(
                                     crate::error::CapturePokemonEventsError::InvalidPokeballError(
                                         crate::error::InvalidPokeballError {
                                             pokeball: pokeball.to_owned()
                                         }
                                     )
-                                );
```

Comment:
Related to Burak's comment, I'm having trouble understanding when the receiver can disconnect and this can panic. Can we document that somewhere?

* If the receiver can indeed drop, we shouldn't have examples where we panic, and instead teach users in the example how best to handle the disconnection.
* If the receiver can't really drop, we shouldn't expose a fallible interface.